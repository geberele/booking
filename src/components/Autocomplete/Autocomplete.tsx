import { IResults } from '../../api/search'
import React from 'react'
import { Item } from '../Item/Item'

interface IAutocomplete  {
  results: IResults[]
  onClick: (value: string) => void
}

export const Autocomplete = ({ results, onClick }: IAutocomplete) => {
  return (
    <>
      {results.length > 0 &&
        <div
          className="autocomplete"
          data-testid="autocomplete"
        >
          <ol className="autocomplete__list">
            {results.map((el, index) => (
              <Item
                key={index}
                result={el}
                onClick={onClick}
              />
            ))}
          </ol>
        </div>
      }
    </>
  )
}
