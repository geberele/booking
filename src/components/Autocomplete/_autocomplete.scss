.autocomplete {
  background: $color-white;
  border-radius: 4px;
  -webkit-box-shadow: 0 1px 2px rgba($color-black, .2);
  box-shadow: 0 1px 2px rgba($color-black, .2);
  overflow: hidden;

  &__list {
    list-style: none;
    margin: 0;
    padding: 0;
  }
}
