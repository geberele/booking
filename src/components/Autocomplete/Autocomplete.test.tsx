import { render } from '@testing-library/react'
import * as React from 'react'
import { Autocomplete, IAutocomplete } from './Autocomplete'

describe('Autocomplete', () => {
  it('renders', () => {
    const props: IAutocomplete = {
      results: [
        {
          country: 'Italy',
          city: 'Milan',
          region: 'Lombardy',
          name: 'Milan Malpensa Airport',
          iata: 'MXP',
          placeType: 'Airport',
        },
        {
          country: 'Italy',
          city: '',
          region: 'Lombardia',
          name: 'Milan',
          iata: '',
          placeType: 'City',
        },
        {
          country: 'Italy',
          city: 'Milan',
          region: 'Lombardy',
          name: 'Milan Linate Airport',
          iata: 'LIN',
          placeType: 'Airport',
        },
        {
          country: 'Italy',
          city: 'Bergamo',
          region: 'Lombardy',
          name: 'Bergamo Airport',
          iata: 'BGY',
          placeType: 'Airport',
        },
        {
          country: 'Italy',
          city: 'Milan',
          region: 'Lombardy',
          name: 'Milan - Central Train Station',
          iata: '',
          placeType: 'Other',
        },
        {
          country: 'Italy',
          city: 'Milano',
          region: 'Lombardy',
          name: 'Gallaratese',
          iata: '',
          placeType: 'Other',
        },
      ],
      onClick: () => {
        return
      },
    }
    const { asFragment } = render(<Autocomplete {...props} />)
    expect(asFragment()).toMatchSnapshot()
  })
})