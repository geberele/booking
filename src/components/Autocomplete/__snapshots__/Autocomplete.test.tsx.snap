// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Autocomplete renders 1`] = `
<DocumentFragment>
  <div
    class="autocomplete"
    data-testid="autocomplete"
  >
    <ol
      class="autocomplete__list"
    >
      <li
        aria-label="Milan Malpensa Airport"
        class="item"
        tabindex="0"
      >
        <div
          class="item__place"
        >
          <span
            class="item__pill"
          >
            Airport
          </span>
        </div>
        <div
          class="item__location"
        >
          <span
            class="item__location-name"
          >
            Milan Malpensa Airport MXP
          </span>
          <span
            class="item__location-city"
          >
            Milan, Lombardy, Italy
          </span>
        </div>
      </li>
      <li
        aria-label="Milan"
        class="item"
        tabindex="0"
      >
        <div
          class="item__place"
        >
          <span
            class="item__pill"
          >
            City
          </span>
        </div>
        <div
          class="item__location"
        >
          <span
            class="item__location-name"
          >
            Milan
          </span>
          <span
            class="item__location-city"
          >
            Lombardia, Italy
          </span>
        </div>
      </li>
      <li
        aria-label="Milan Linate Airport"
        class="item"
        tabindex="0"
      >
        <div
          class="item__place"
        >
          <span
            class="item__pill"
          >
            Airport
          </span>
        </div>
        <div
          class="item__location"
        >
          <span
            class="item__location-name"
          >
            Milan Linate Airport LIN
          </span>
          <span
            class="item__location-city"
          >
            Milan, Lombardy, Italy
          </span>
        </div>
      </li>
      <li
        aria-label="Bergamo Airport"
        class="item"
        tabindex="0"
      >
        <div
          class="item__place"
        >
          <span
            class="item__pill"
          >
            Airport
          </span>
        </div>
        <div
          class="item__location"
        >
          <span
            class="item__location-name"
          >
            Bergamo Airport BGY
          </span>
          <span
            class="item__location-city"
          >
            Bergamo, Lombardy, Italy
          </span>
        </div>
      </li>
      <li
        aria-label="Milan - Central Train Station"
        class="item"
        tabindex="0"
      >
        <div
          class="item__place"
        >
          <span
            class="item__pill"
          >
            Other
          </span>
        </div>
        <div
          class="item__location"
        >
          <span
            class="item__location-name"
          >
            Milan - Central Train Station
          </span>
          <span
            class="item__location-city"
          >
            Milan, Lombardy, Italy
          </span>
        </div>
      </li>
      <li
        aria-label="Gallaratese"
        class="item"
        tabindex="0"
      >
        <div
          class="item__place"
        >
          <span
            class="item__pill"
          >
            Other
          </span>
        </div>
        <div
          class="item__location"
        >
          <span
            class="item__location-name"
          >
            Gallaratese
          </span>
          <span
            class="item__location-city"
          >
            Milano, Lombardy, Italy
          </span>
        </div>
      </li>
    </ol>
  </div>
</DocumentFragment>
`;
